test/e2e/tas/leaderworkerset_test.go

```diff
@@ -0,0 +1,184 @@
+								kueuealpha.PodSetRequiredTopologyAnnotation: testing.DefaultBlockTopologyLevel,
+							},
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "c",
+									Image: util.E2eTestSleepImage,
+									Args:  []string{"10m"},
+									Resources: corev1.ResourceRequirements{
+										Limits: map[corev1.ResourceName]resource.Quantity{
+											extraResource: resource.MustParse("1"),
+										},
+										Requests: map[corev1.ResourceName]resource.Quantity{
+											extraResource: resource.MustParse("1"),
+										},
+									},
+								},
+							},
+						},
+					}).
+					Obj()
+				ginkgo.By("Create a LeaderWorkerSet", func() {
+					gomega.Expect(k8sClient.Create(ctx, lws)).To(gomega.Succeed())
+				})
+
+				ginkgo.By("Waiting for replicas is ready", func() {
+					ginkgo.By("Waiting for replicas is ready", func() {
+						createdLeaderWorkerSet := &leaderworkersetv1.LeaderWorkerSet{}
+						gomega.Eventually(func(g gomega.Gomega) {
+							g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(lws), createdLeaderWorkerSet)).To(gomega.Succeed())
+							g.Expect(createdLeaderWorkerSet.Status.ReadyReplicas).To(gomega.Equal(replicas))
+							g.Expect(createdLeaderWorkerSet.Status.Conditions).To(gomega.ContainElement(
+								gomega.BeComparableTo(metav1.Condition{
+									Type:    "Available",
+									Status:  metav1.ConditionTrue,
+									Reason:  "AllGroupsReady",
+									Message: "All replicas are ready",
+								}, util.IgnoreConditionTimestampsAndObservedGeneration)),
+							)
+						}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+					})
+				})
+
+				pods := &corev1.PodList{}
+				ginkgo.By("ensure all pods are scheduled", func() {
+					listOpts := &client.ListOptions{
+						FieldSelector: fields.OneTermNotEqualSelector("spec.nodeName", ""),
+					}
+					gomega.Eventually(func(g gomega.Gomega) {
+						g.Expect(k8sClient.List(ctx, pods, client.InNamespace(ns.Name), listOpts)).To(gomega.Succeed())
+						g.Expect(pods.Items).Should(gomega.HaveLen(int(podsTotalCount)))
+					}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+				})
+
+				ginkgo.By("verify the assignment of pods are as expected with rank-based ordering", func() {
+					gomega.Expect(k8sClient.List(ctx, pods, client.InNamespace(ns.Name))).To(gomega.Succeed())
+					gotAssignment := make(map[string]string, podsTotalCount)
+					for _, pod := range pods.Items {
+						index := fmt.Sprintf("%s/%s", pod.Labels[leaderworkersetv1.GroupIndexLabelKey], pod.Labels[leaderworkersetv1.WorkerIndexLabelKey])
+						gotAssignment[index] = pod.Spec.NodeName
+					}
+					gomega.Expect(gotAssignment).Should(gomega.BeComparableTo(wantAssignment))
+				})
+			},
+			ginkgo.Entry(
+				"one leader",
+				int32(1),
+				int32(3),
+				map[string]string{
+					"0/0": "kind-worker",
+					"0/1": "kind-worker2",
+					"0/2": "kind-worker3",
+				},
+			),
```

Comment:
What this case adds additionally in terms of coverage? If there is no benefit I would suggest to drop it and don't use "DesribeTable". It makes the test harder to follow.

Reply:
Done